curvefs/src/client/inode_cache_manager.cpp

```diff
@@ -183,47 +213,36 @@ void InodeCacheManagerImpl::FlushInodeOnce() {
 void InodeCacheManagerImpl::AddParent(uint64_t inodeId, uint64_t parentId) {
     curve::common::LockGuard lg2(parentIdMapMutex_);
     if (parentIdMap_.count(inodeId)) {
-        parentIdMap_[inodeId].emplace_back(parentId);
-    } else {
-        parentIdMap_.emplace(inodeId, std::list<uint64_t>({parentId}));
-    }
-}
-
-void InodeCacheManagerImpl::RemoveParent(uint64_t inodeId, uint64_t parentId) {
-    curve::common::LockGuard lg2(parentIdMapMutex_);
-    if (parentIdMap_.count(inodeId)) {
-        auto iter = std::find(parentIdMap_[inodeId].begin(),
-            parentIdMap_[inodeId].end(), parentId);
-        if (iter != parentIdMap_[inodeId].end()) {
-            parentIdMap_[inodeId].erase(iter);
-        }
+        LOG(WARNING) << "AddParent but exist, inodeId = " << inodeId
+                     << ", parentId = " << parentIdMap_[inodeId]
+                     << ", need2AddParentId = " << parentId;
```

Comment:
check old parent and new parent are equal?

Reply:
> check old parent and new parent are equal?

fixed

Reply:
if (parentIdMap_.count(inodeId) && parentIdMap_[inodeId] != parentId) 
what happens here?

Reply:
> if (parentIdMap_.count(inodeId) && parentIdMap_[inodeId] != parentId) what happens here?

It won't happen normally, just do anomaly detection.